src/oss-characteristics/CharacteristicTool.cs

```diff
@@ -193,7 +193,7 @@ private static List<string> GetTextResults(PackageURL purl, Dictionary<string, A
                     stringOutput.Add(string.Format("Programming Language: {0}",
                         string.Join(", ", metadata?.Languages?.Keys ?? Array.Empty<string>().ToList())));
                     stringOutput.Add("Unique Tags: ");
-                    foreach (var tag in metadata?.UniqueTags ?? new ConcurrentDictionary<string, byte>())
+                    foreach (var tag in metadata?.UniqueTags.Keys ?? new ConcurrentDictionary<string, byte>().Keys)
```

Comment:
```suggestion
                    foreach (var tag in metadata?.UniqueTags?.Keys ?? Array.Empty<string>())
```

Reply:
Recommend a similar change on line 165 for the sarif result output where we are writing out the byte value.